### PR TITLE
add centripetal head

### PR DESCRIPTION
**Pull request by tianyuandu**

### CODE REVIEW

File: mmdet/models/dense_heads/centripetal_head.py [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+from ..builder import HEADS, build_loss
+from .corner_head import CornerHead
+
+
+@HEADS.register_module()
+class CentripetalHead(CornerHead):
+    """Head of CentripetalNet: Pursuing High-quality Keypoint Pairs for Object
+    Detection.
+
+    CentripetalHead is inherit from :class:`CornerHead`. It removes the
```
Comment (ZwwWayne) [inline mmdet/models/dense_heads/centripetal_head.py:15]: CentripetalHead inherits from :class:`CornerHead`.

File: mmdet/models/dense_heads/centripetal_head.py [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+    embedding branch and adds guiding shift and centripetal shift branches.
+    More details can be found in the `paper
+    <https://arxiv.org/abs/2003.09119>`_ .
+
+    Args:
+        num_classes (int): Number of categories excluding the background
+            category.
+        in_channels (int): Number of channels in the input feature map.
+        num_feat_levels (int): Levels of feature from the previous module. 2
+            for HourglassNet-104 and 1 for HourglassNet-52. Because
```
Comment (ZwwWayne) [inline mmdet/models/dense_heads/centripetal_head.py:25]: `Because` is unnecessary here.

File: mmdet/models/dense_heads/centripetal_head.py [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+    """
+
+    def __init__(self,
+                 *args,
+                 loss_guiding_shift=dict(
+                     type='SmoothL1Loss', beta=1.0, loss_weight=0.05),
+                 loss_centripetal_shift=dict(
+                     type='SmoothL1Loss', beta=1.0, loss_weight=1),
+                 **kwargs):
+        self.centripetal_shift_channels = 2
```
Comment (ZwwWayne) [inline mmdet/models/dense_heads/centripetal_head.py:52]: Can we set these three attributes through argument and set their default values?

File: mmdet/models/dense_heads/centripetal_head.py [hunk 4]
```diff
@@ -0,0 +1,396 @@
...
+        self.loss_centripetal_shift = build_loss(loss_centripetal_shift)
+
+    def _init_centripetal_layers(self):
+        """Initialize centripetal layers. Including feature adaption deform
+        convs (fadp), deform offset prediction convs (dcn_off), guiding shift
+        (guiding_shift) and centripetal shift (centripetal_shift). Each branch
+        has two parts: prefix `tl_` for top-left and `br_` for bottom-right.
+        """
+        self.tl_fadp, self.br_fadp = nn.ModuleList(), nn.ModuleList()
+        self.tl_dcn_off, self.br_dcn_off = nn.ModuleList(), nn.ModuleList()
```
Comment (ZwwWayne) [inline mmdet/models/dense_heads/centripetal_head.py:66]: `off` is ambiguous, `tl_dcn_offset` and `br_dcn_offset` might be better

File: mmdet/models/dense_heads/centripetal_head.py [hunk 5]
```diff
@@ -0,0 +1,396 @@
...
+        self.loss_guiding_shift = build_loss(loss_guiding_shift)
+        self.loss_centripetal_shift = build_loss(loss_centripetal_shift)
+
+    def _init_centripetal_layers(self):
+        """Initialize centripetal layers. Including feature adaption deform
+        convs (fadp), deform offset prediction convs (dcn_off), guiding shift
+        (guiding_shift) and centripetal shift (centripetal_shift). Each branch
+        has two parts: prefix `tl_` for top-left and `br_` for bottom-right.
+        """
+        self.tl_fadp, self.br_fadp = nn.ModuleList(), nn.ModuleList()
```
Comment (ZwwWayne) [inline mmdet/models/dense_heads/centripetal_head.py:65]: What does `fadp` mean? `feature adaptation`? Try a more clear name for that.

**Comment by hellock** [comment]
Task linked: [CU-4fprqd Centripetal Head](https://app.clickup.com/t/4fprqd)

**Review by ZwwWayne — APPROVED** [review]

**Review by ZwwWayne — APPROVED** [review]

**Comment by hellock** [comment]
README and modelzoo.md should also be updated.

**Pull request merged**